### PR TITLE
Add documentation for too-many-lines

### PR DESCRIPTION
**Pull request by ollie-iterators**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Add documentation for too-many-lines.

Refs #7897 

Closes #XXXX

### CODE REVIEW

File: doc/data/messages/d/duplicate-code/details.rst [hunk 1]
```diff
@@ -1,9 +0,0 @@
-If you need to make a change to the logic or functionality of the duplicated
-code, you will need to identify all the places that need to be changed, which
-can be time-consuming and error-prone. If there are multiple copies of the
-same code, then you will also need to test each copy to ensure that the
-functionality is correct. Duplicate code can be confusing for someone who is
-trying to understand the logic and flow of the code if they come across multiple
-identical or nearly identical blocks of code. The reader can then skim and
-think something is identical when it actually isn't. This is particularly true
-during review.
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/duplicate-code/details.rst:9]: I think this was removed by mistake.

File: doc/data/messages/t/too-many-lines/bad/apple.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    def take_bite(self):
+        if self.remaining_bites > 0:
+            print("You take a bite of the apple.")
+            self.remaining_bites -= 1
+        else:
+            print("The apple is already eaten up!")
+
+    def eaten_by_animal(self, animal):
+        self.remaining_bites = 0
+        print("The apple has been eaten by an animal.")
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/bad/apple.py:14]: We need a single big file with two things in bad.py and two file with one thing in each in good/

File: doc/data/messages/t/too-many-lines/details.rst [hunk 3]
```diff
@@ -1 +0,0 @@
-You can help us make the doc better `by contributing <https://github.com/PyCQA/pylint/issues/5953>`_ !
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/details.rst:1]: Details should contain a rational for why file with too much lines are bad.

File: pylint/message/_deleted_message_ids.py [hunk 4]
```diff
@@ -123,6 +123,9 @@ class DeletedMessage(NamedTuple):
     "https://github.com/PyCQA/pylint/pull/6421": [
         DeletedMessage("W0111", "assign-to-new-keyword"),
     ],
+    "https://github.com/PyCQA/pylint/pull/8323": [
+        DeletedMessage("W2402", "non-ascii-file-name"),
+    ],
```
Comment (Pierre-Sassoulas) [inline pylint/message/_deleted_message_ids.py:128]: Let's do that in another MR

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you for working on this, looks promising.

File: doc/data/messages/d/duplicate-code/details.rst (hunk 1 above)
Comment (ollie-iterators) [reply doc/data/messages/d/duplicate-code/details.rst]: Fixed

File: doc/data/messages/t/too-many-lines/bad/apple.py (hunk 2 above)
Comment (ollie-iterators) [reply doc/data/messages/t/too-many-lines/bad/apple.py]: Made one file

File: doc/data/messages/t/too-many-lines/details.rst (hunk 3 above)
Comment (ollie-iterators) [reply doc/data/messages/t/too-many-lines/details.rst:1]: Added some words to details.rst

File: pylint/message/_deleted_message_ids.py (hunk 4 above)
Comment (ollie-iterators) [reply pylint/message/_deleted_message_ids.py]: Reverted

File: doc/data/messages/t/too-many-lines/good.py [hunk 5]
```diff
@@ -1 +1,14 @@
...
+def isPalindrome(string):
+	left_pos = 0
+	right_pos = len(string) - 1
+
+	while right_pos >= left_pos:
+		if not string[left_pos] == string[right_pos]:
+			return False
+		left_pos += 1
+		right_pos -= 1
+	return True
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/good.py]:
```suggestion
def is_palindrome(s):
    return s == s[::-1]
```

File: doc/data/messages/t/too-many-lines/good.py [hunk 6]
```diff
@@ -1 +1,14 @@
...
+
+listOfStrings = ['aza', 'racecar', 'trigger', 'ogre']
+for string in listOfStrings:
+    print(isPalindrome(string))
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/good.py]:
```suggestion
strings = ['aza', 'racecar', 'trigger', 'ogre']
for string in strings:
    print(is_palindrome(string))
```

File: doc/data/messages/t/too-many-lines/bad.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+def isPalindrome(string): # [too-many-lines]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/bad.py:1]:
```suggestion
def is_palindrome(string): # [too-many-lines]
```

File: doc/data/messages/t/too-many-lines/bad.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+	left_pos = 0
+	right_pos = len(string) - 1
+
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/bad.py:4]:
```suggestion
```

File: doc/data/messages/t/too-many-lines/bad.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+			return False
+		left_pos += 1
+		right_pos -= 1
+	return True
+
+
+print(isPalindrome('aza'))
+print(isPalindrome('racecar'))
+print(isPalindrome('trigger'))
+print(isPalindrome('ogre'))
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/bad.py]:
```suggestion
print(is_palindrome('aza'))
print(is_palindrome('racecar'))
print(is_palindrome('trigger'))
print(is_palindrome('ogre'))
```

File: doc/data/messages/t/too-many-lines/details.rst [hunk 10]
```diff
@@ -1 +1,3 @@
...
+When a module has too many lines it can make it harder for the module to run because too much code is
+in the module. Also it makes it hard to find the code that you are looking for because what you are
+looking for could be happing multiple times in the module.
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/details.rst:3]:
```suggestion
When a module has too many lines it can make it difficult to read and understand. There might be
performance issue while editing the file because the IDE must parse more code. You need more expertise
to navigate the file properly (go to a particular line when debugging, or search for a specific code construct, instead of navigating by clicking and scrolling)

This measure is a proxy for higher cyclomatic complexity that you might not be calculating if you're not using ``load-plugins=pylint.extensions.mccabe,``. Cyclomatic complexity is slower to compute, but also a more fine grained measure than raw SLOC. In particular, you can't make the code less readable by making a very complex one liner if you're using cyclomatic complexity.

The example simplify the code, but it's not always possible. Most of the time bursting the file 
by creating a package with the same API is the only solution. Anticipating and creating the file
from the get go will permit to have the same end result with a better version control history.
```

File: doc/data/messages/t/too-many-lines/good/is_palindrome.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
+def isPalindrome(string):
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/good/is_palindrome.py:1]:
```suggestion
def is_palindrome(string):
```

File: doc/data/messages/t/too-many-lines/good/main.py [hunk 12]
```diff
@@ -0,0 +1,4 @@
+def main():
+	listOfStrings = ['aza', 'racecar', 'trigger', 'ogre']
+	for string in listOfStrings:
+		print(isPalindrome(string))
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/too-many-lines/good/main.py:4]:
```suggestion
def main():
	strings = ['aza', 'racecar', 'trigger', 'ogre']
	for string in strings:
		print(is_palindrome(string))
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you @ollie-iterators !
